assignment4/Q1.cpp

```diff
@@ -0,0 +1,126 @@
+#include <iostream>
+#include <vector>
+#include <set>
+
+using namespace std;
+//iterating two times - first to copy the board
+//not sure of its correctness
+//probably something is missing but I do not know yet
+
+struct Point {
+    int x;
+    int y;
+    char val;
+    int set_no;
+};
+
+//returns the set whom point belongs to
+int findSet(struct Point point) {
+    return point.set_no;
+}
+
+void unionSets(set<int> &sets, int set_to_erase, int set_to_stay) {
+
+    sets.erase(set_to_erase);
+    return;
+}
+
+int count_islands(vector <vector<char> > board) {
+
+    int height = board.size();
+    int width = board[0].size();
+    int set_no = 0;
+    set<int> sets; //for storing sets' numbers
+
+    //initialize board with guards
+    struct Point template_point;
```

Comment:
Right, that's unfortunate mistake. I've been using much C lately and got used to writing "struct" everytime. The same with "T" and "F" char values - I forgot about booleans in C++ for a moment.